File: kernels/C++/compositron/examples/dbs_fit.cpp
#include <iostream>

#include "compositron/core/measurement.hpp"
#include "compositron/dbs/dbspectrum.hpp"


int main() {
    compositron::core::Measurement m(
        "../../../../testdata/depth-profile_Copper_0000.n42"
    );

    std::cout << m.shape() << std::endl;

    auto& d = m.dbs.at("OAB");

    d.default_analyze();

    auto s = d.lineshape_params["S"];
    auto w = d.lineshape_params["W"];
    auto v = d.lineshape_params["V/P"];
    auto p = d.lineshape_params["P/T"];

    std::cout << "  S:   " << s.val << " +/- " << s.err << std::endl;
    std::cout << "  W:   " << w.val << " +/- " << w.err << std::endl;
    std::cout << "  V/P: " << v.val << " +/- " << v.err << std::endl;
    std::cout << "  P/T: " << p.val << " +/- " << p.err << std::endl;

    return 0;
}

File: kernels/Python/compositron/examples/dbs_fit.py
from compositron.core.measurement import Measurement


m = Measurement(
    "../../../../testdata/depth-profile_Copper_0000.n42"
)

print(m.shape)

d = m.dbs["OAB"]

d.default_analyze()

s = d.lineshape_params["S"]
w = d.lineshape_params["W"]
v = d.lineshape_params["V/P"]
p = d.lineshape_params["P/T"]

print(f"  S     = {s.val:.6f} +/- {s.err:.6f}")
print(f"  W     = {w.val:.6f} +/- {w.err:.6f}")
print(f"  V/P   = {v.val:.6f} +/- {v.err:.6f}")
print(f"  P/T   = {p.val:.6f} +/- {p.err:.6f}")
